binder/binder.cpp

```diff
@@ -0,0 +1,1150 @@
+            break;
+        }
+        *status = GBINDER_STATUS_OK;
+        return gbinder_local_reply_append_int32(gbinder_local_object_new_reply(obj), 0);
+    } else {
+        qWarning("Unknown interface %s and code %u\n", iface, code);
+        *status = GBINDER_STATUS_FAILED;
+    }
+    return NULL;
+}
+
+
+static GBinderLocalReply* geoclue_binder_agnss_ril_callback(
+    GBinderLocalObject* obj,
+    GBinderRemoteRequest* req,
+    guint code,
+    guint flags,
+    int* status,
+    void* user_data)
+{
+    Q_UNUSED(flags)
+    Q_UNUSED(user_data)
+    const char* iface = gbinder_remote_request_interface(req);
+
+    if (!g_strcmp0(iface, AGNSS_RIL_CALLBACK)) {
+        GBinderReader reader;
+
+        gbinder_remote_request_init_reader(req, &reader);
+        switch (code) {
+        case AGNSS_RIL_REQUEST_REF_ID_CB:
+            qCDebug(lcGeoclueHybris) << "AGNSS RIL request ref ID";
+            break;
+        case AGNSS_RIL_REQUEST_REF_LOC_CB:
+            qCDebug(lcGeoclueHybris) << "AGNSS RIL request ref location";
+            break;
+        default:
+            qWarning("Failed to decode callback %u\n", code);
+            break;
+        }
+        *status = GBINDER_STATUS_OK;
+        return gbinder_local_reply_append_int32(gbinder_local_object_new_reply(obj), 0);
+    } else {
+        qWarning("Unknown interface %s and code %u\n", iface, code);
+        *status = GBINDER_STATUS_FAILED;
+    }
+    return NULL;
+}
+
+
+static GBinderLocalReply* geoclue_binder_gnss_ni_callback(
+    GBinderLocalObject* obj,
+    GBinderRemoteRequest* req,
+    guint code,
+    guint flags,
+    int* status,
+    void* user_data)
+{
+    Q_UNUSED(flags)
+    Q_UNUSED(user_data)
+    const char* iface = gbinder_remote_request_interface(req);
+
+    if (!g_strcmp0(iface, GNSS_NI_CALLBACK)) {
+        GBinderReader reader;
+
+        gbinder_remote_request_init_reader(req, &reader);
+        switch (code) {
+        case GNSS_NI_NOTIFY_CB:
+            qCDebug(lcGeoclueHybris) << "GNSS NI notify";
+            break;
+        default:
+            qWarning("Failed to decode callback %u\n", code);
+            break;
+        }
+        *status = GBINDER_STATUS_OK;
+        return gbinder_local_reply_append_int32(gbinder_local_object_new_reply(obj), 0);
+    } else {
+        qWarning("Unknown interface %s and code %u\n", iface, code);
+        *status = GBINDER_STATUS_FAILED;
+    }
+    return NULL;
+}
+
+static void geoclue_binder_gnss_gnss_died(
+    GBinderRemoteObject */*obj*/,
+    void* user_data)
+{
+    HybrisBinder* self = (HybrisBinder *)user_data;
+    self->dropGnss();
+}
+
+
+/*==========================================================================*
+ * Backend class
+ *==========================================================================*/
+
+HybrisBinder::HybrisBinder(QObject *parent)
+:   QObject(parent), m_death_id(0), m_fqname(0), m_sm(0),
+    m_clientGnss(0),
```

Comment:
nitpick: could use Q_NULLPTR instead of literal 0 for pointer initialization

Reply:
The original code used 0 for null pointer so I used the same to make the code consistent. Maybe we should replace similar values also in the old code with Q_NULLPTR?